x-pack/legacy/plugins/kuery_autocomplete/public/autocomplete_providers/index.js

```diff
@@ -22,7 +22,7 @@ export const kueryProvider = ({ config, indexPatterns, boolFilter }) => {
     return provider({ config, indexPatterns, boolFilter });
   });
 
-  return function getSuggestions({ query, selectionStart, selectionEnd }) {
+  return function getSuggestions({ query, selectionStart, selectionEnd, signal }) {
```

Comment:
// nit
Can\will the signal be used for anything but aborting?
If not, maybe call it abortSignal?